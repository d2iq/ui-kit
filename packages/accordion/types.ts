export type HeadingLevel = 2 | 3 | 4 | 5 | 6;
export type AccordionTitleAppearances = "default" | "danger";
