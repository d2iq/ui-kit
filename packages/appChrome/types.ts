import { BreakpointConfig } from "../shared/styles/breakpoints";
import { SpaceSizes } from "../shared/styles/styleUtils/modifiers/modifierUtils";
import { LinkProps } from "../link/types";
import { IconSize } from "../shared/types/iconSize";

type BgColor = React.CSSProperties["backgroundColor"];
type PaddingHoriz = SpaceSizes;
type PaddingVert = SpaceSizes;

export interface AppChromeTheme {
  headerBackgroundColor?: BgColor;
  headerPaddingHor?: PaddingHoriz;
  headerPaddingVert?: PaddingVert;
  iconWidth?: IconSize;
  itemActiveBackgroundColor?: BgColor;
  itemHoverBackgroundColor?: BgColor;
  sidebarBackgroundColor?: BgColor;
  sidebarHeaderPaddingHor?: PaddingHoriz;
  sidebarHeaderPaddingVert?: PaddingVert;
  sidebarWidth?: BreakpointConfig<React.CSSProperties["width"]>;
  sidebarItemPaddingHor?: PaddingHoriz;
  sidebarItemPaddingVert?: PaddingVert;
}

export interface SidebarNavItemProps extends LinkProps {
  isActive?: boolean;
  onClick?: (event?: React.SyntheticEvent<HTMLElement>) => void;
  disabled?: boolean;
}
