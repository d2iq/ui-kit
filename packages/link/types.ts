import React from "react";

export interface LinkProps {
  /**
   * A url the user will be navigated to when clicking the link. This also changes the tag to `<a>`.
   */
  url?: string;
  /**
   * If the `url` prop is set, this will open that link in a new tab.
   */
  openInNewTab?: boolean;
  children?: React.ReactNode | React.ReactNode[];
}

export type ExpandedLinkProps = LinkProps &
  Omit<React.HTMLProps<HTMLAnchorElement>, "ref">;

export type LinkComponent = React.ComponentType<
  LinkProps & React.HTMLProps<HTMLAnchorElement>
>;
