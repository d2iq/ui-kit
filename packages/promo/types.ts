export interface PromoProps {
  /** The primary banner content */
  bodyContent: React.ReactNode;
  /** A function that gets called when the banner is dismissed */
  dismissHandler?: () => void;
  /** A consise summary of the banner's message */
  headingText: string;
  /** Whether the content is on a dark background. Changes text color to be readable. */
  isDarkBackground?: boolean;
  /** A link to the graphic's image file. The graphic (typically an illustration) should be relevant to the banner's message. Ideally the graphic should not just be decorative, but it reinforces the banner's overall message. */
  graphicSrc?: string;
  /** Whether a user has opted not to see the banner */
  optOutBanner?: boolean;
  /** A function that gets called when the user checks the "Do not show again" checkbox */
  optOutHandler?: (e: React.SyntheticEvent<HTMLInputElement>) => void;
  /** The most relevant action a user can take in response to the banner's message */
  primaryAction?: React.ReactNode;
  /** An additional action a user can take in response to the banner's message. This can also just be a link to more information relevant to the banner's message. */
  secondaryAction?: React.ReactNode;
  /** The Cypress selector */
  ["data-cy"]?: string;
}

export type GradientStyle = "lightBlue" | "purple" | "oceanBlue";
export type GradientColors = [string, string, string?];
export type PromoBackgroundColor = "themeBgSecondary" | "purpleLighten5";
