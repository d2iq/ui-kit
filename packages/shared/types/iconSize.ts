export type IconSize = "xxs" | "xs" | "s" | "m" | "l" | "xl" | "xxl";
