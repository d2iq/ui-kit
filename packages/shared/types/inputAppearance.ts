export enum InputAppearance {
  Standard = "standard",
  Error = "error",
  Success = "success"
}
