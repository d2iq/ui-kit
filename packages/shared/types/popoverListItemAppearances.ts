export enum PopoverListItemAppearances {
  DANGER = "danger"
}
