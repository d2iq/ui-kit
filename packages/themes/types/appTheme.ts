import { AppChromeTheme } from "../../appChrome/types";

type Color = React.CSSProperties["color"];
export type LocalTheme = AppChromeTheme & {
  menuHasIcon: boolean;

  coloredRows: any[];
  mutedRows: any[];

  colors: {
    // Branding
    brandPrimary?: Color;
    brandPrimaryInverted?: Color;
    brandSecondary?: Color;
    brandSecondaryInverted?: Color;

    // States
    success?: Color;
    successInverted?: Color;
    warning?: Color;
    warningInverted?: Color;
    error?: Color;
    errorInverted?: Color;

    // Backgrounds
    bgPrimary?: Color;
    bgPrimaryInverted?: Color;
    bgSecondary?: Color;
    bgSecondaryInverted?: Color;
    bgHover?: Color;
    bgHoverInverted?: Color;
    bgScrim?: Color;
    bgSelected?: Color;
    bgSelectedInverted?: Color;
    bgAppHeader?: Color;

    // Text
    textColorPrimary?: Color;
    textColorPrimaryInverted?: Color;
    textColorSecondary?: Color;
    textColorSecondaryInverted?: Color;
    textColorInteractive?: Color;
    textColorInteractiveInverted?: Color;

    // Decorators
    border?: Color;
    borderInverted?: Color;
    borderHeavy?: Color;
    borderHeavyInverted?: Color;
    shadow?: Color;
    shadowInverted?: Color;
  };
};

declare module "@emotion/react" {
  export interface Theme extends LocalTheme {
    _unused: true; // can not be empty
  }
}
