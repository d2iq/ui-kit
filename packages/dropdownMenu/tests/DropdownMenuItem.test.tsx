import React from "react";
import { mount } from "enzyme";
import { createSerializer } from "@emotion/jest";
import toJson from "enzyme-to-json";
import { PopoverListItemAppearances } from "../../shared/types/popoverListItemAppearances";

import { DropdownMenuItem } from "..";

expect.addSnapshotSerializer(createSerializer());

describe("Dropdown", () => {
  it("default", () => {
    const component = mount(
      <DropdownMenuItem key="edit" value="edit">
        Edit
      </DropdownMenuItem>
    );

    expect(toJson(component)).toMatchSnapshot();
  });
  it("default", () => {
    Object.keys(PopoverListItemAppearances).forEach(appearance => {
      const component = mount(
        <DropdownMenuItem
          key="edit"
          value="edit"
          appearance={PopoverListItemAppearances[appearance]}
        >
          Edit
        </DropdownMenuItem>
      );

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
