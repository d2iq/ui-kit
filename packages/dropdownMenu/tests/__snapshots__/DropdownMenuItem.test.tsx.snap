// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown default 1`] = `
<DropdownMenuItem
  value="edit"
>
  Edit
</DropdownMenuItem>
`;

exports[`Dropdown default 2`] = `
<DropdownMenuItem
  appearance="danger"
  value="edit"
>
  Edit
</DropdownMenuItem>
`;
